app/models/cart/validator.rb

```diff
@@ -0,0 +1,26 @@
+module Shop
+  module Models
+    class Cart
+      include Exceptions
+
+      class Validator
+        def initialize(cart)
+          @cart = cart
+        end
+
+        def validate!
+          cart_items_only?
+        end
+
+        private
+
+        def cart_items_only?
+          raise Cart::InvalidCartItemsTypeError unless @cart.items.is_a?(Array)
+          @cart.items.each do |item|
+            raise Cart::InvalidCartItemsTypeError unless item.is_a?(CartItem)
+          end
```

Comment:
Wow. That was dumb of me. Thanks for pointing this out. I'm not really used to having so many actually helpful methods to choose from.